package com.ecommerce.adapter.in.web.cart.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CartDTO {
    private String id;
    @Builder.Default
    private List<CartItemDTO> items = new ArrayList<>();
    private Integer totalItems;
    private BigDecimal totalAmount;
}

